mne/channels/interpolation.py

```diff
@@ -213,10 +239,6 @@ def _interpolate_bads_meeg(
         # select the bad channels to be interpolated
         picks_bad = pick_channels(inst.info["ch_names"], bads_type, exclude=[])
 
-        if method[ch_type] == "nan":
-            inst._data[picks_bad] = np.nan
```

Comment:
This is the line with the bug as far as I can tell

Reply:
As opposed to here https://github.com/mne-tools/mne-python/pull/12336/files#diff-fcfbd5c9390aa74bdd3005e5ae1c786df72afc324d3532273dc5937ef9674787L130-R131 which does it correctly